packages/react-storage/src/components/StorageBrowser/actions/handlers/listLocationItems.ts

```diff
@@ -42,4 +48,32 @@ export interface ListLocationItemsHandler
   > {}
 
 export const listLocationItemsHandler: ListLocationItemsHandler =
-  null as unknown as ListLocationItemsHandler;
+  async function listLocationItemsAction(
+    input: ListLocationItemsHandlerInput
+  ): Promise<ListLocationItemsHandlerOutput> {
+    const { config, options, prefix: path } = input ?? {};
+    const { delimiter, nextToken, pageSize } = options ?? {};
+
+    const { accountId, bucket: bucketName, region } = config ?? {};
+
+    const bucket = { bucketName, region };
+    const subpathStrategy: StorageSubpathStrategy = {
+      delimiter,
+      strategy: delimiter ? 'exclude' : 'include',
+    };
+
+    const listInput: ListPaginateInput = {
+      path,
+      options: {
+        bucket,
+        expectedBucketOwner: accountId,
+        nextToken,
+        pageSize,
+        subpathStrategy,
+      },
+    };
+
+    const output = await list(listInput);
```

Comment:
We are missing logic that is built in to the [current handler](https://github.com/aws-amplify/amplify-ui/blob/feat-storage-browser/main/packages/react-storage/src/components/StorageBrowser/do-not-import-from-here/actions/listLocationItemsAction.ts#L80) that we could extend for search recursion

Reply:
Plan to remove this handler from the PR. @reesscot 's [PR](https://github.com/aws-amplify/amplify-ui/pull/5982) will add the new handler.